UefiCpuPkg/Library/MpInitLib/DxeMpLib.c

```diff
@@ -34,6 +47,65 @@ UINTN                   mApPageTable;
 // Begin wakeup buffer allocation below 0x88000
 //
 STATIC EFI_PHYSICAL_ADDRESS  mSevEsDxeWakeupBuffer = 0x88000;
+// MU_CHANGE START: Install protocol for reporting multi-processor debug info
+
+/**
+  Add CPU_MP_DEBUG_PROTOCOL entry to the global list
+
+  @param[in]  StackBuffer      Start of AP stack buffer
+  @param[in]  StackSize        Size of the stack
+  @param[in]  CpuNumber        AP CPU number
+  @param[in]  IsSwitchStack    If the input buffer is the CPU switch stack
+**/
+VOID
+EFIAPI
+AppendCpuMpDebugProtocolEntry (
+  UINTN    StackBuffer,
+  UINTN    StackSize,
+  UINTN    CpuNumber,
+  BOOLEAN  IsSwitchStack
+  )
+{
+  CPU_MP_DEBUG_PROTOCOL  *Entry;
+
+  Entry = AllocatePool (sizeof (CPU_MP_DEBUG_PROTOCOL));
+  if (Entry == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a - Failed to allocate memory!\n", __FUNCTION__));

```

Comment:
Something to consider.

```suggestion
    DEBUG ((DEBUG_ERROR, "%a - Failed to allocate memory!\n", __func__));
```